R/io.R

```diff
@@ -386,3 +386,95 @@ get_land_category <- function(percent_area){
   }
 
 }
+
+#' get_demeter_file
+#' @param irrigation_file_path FUll path to the demeter data file
+#' @details Load demeter file for irrigation and rainfed crop data
+#' @import vroom
+#' @importFrom sp SpatialPointsDataFrame
+#' @author Kristian Nelson (kristian.nelson@pnnl.gov)
+#' @export
+
+get_demeter_file <- function(irrigation_file_path){
+
+  vroom(irrigation_file_path,
+        col_types = cols(
+          latitude = col_double(),
+          longitude = col_double(),
+          corn_irr = col_double(),
+          fibercrop_irr = col_double(),
+          foddergrass_irr = col_double(),
+          fodderherb_irr = col_double(),
+          misccrop_irr = col_double(),
+          oilcrop_irr = col_double(),
+          othergrain_irr = col_double(),
+          palmfruit_irr = col_double(),
+          rice_irr = col_double(),
+          root_tuber_irr = col_double(),
+          sugarcrop_irr = col_double(),
+          wheat_irr = col_double(),
+          corn_rfd = col_double(),
+          fibercrop_rfd = col_double(),
+          foddergrass_rfd = col_double(),
+          fodderherb_rfd = col_double(),
+          misccrop_rfd = col_double(),
+          oilcrop_rfd = col_double(),
+          othergrain_rfd = col_double(),
+          palmfruit_rfd = col_double(),
+          rice_rfd = col_double(),
+          root_tuber_rfd = col_double(),
+          sugarcrop_rfd = col_double(),
+          wheat_rfd = col_double(),
+          area = col_double())) -> demeter
+  # convert to spatial points so that it can be masked by watershed.
+  SpatialPointsDataFrame(coords = demeter[ ,c(2,1)], demeter, proj4string = CRS("+proj=longlat +datum=WGS84 +no_defs")) -> usa_irrigation
+
+  return(usa_irrigation)
+}
+
+#' get_irrigation_count
+#' @param irrigation_city irrigation data subsetted by city watersheds
+#' @details Count irrigated crops vs rainfed crops and determine irrigation status
+#' @importFrom tibble enframe
+#' @importFrom dplyr mutate case_when filter if_else
+#' @author Kristian Nelson (kristian.nelson@pnnl.gov)
+#' @export
+
+get_irrigation_count <- function(irrigation_city){
+
+  # add values of all columns to get sum (in meters) of area irrigated and rainfed.
+  colSums(irrigation_city[,c(3:26)]) %>% enframe() -> dem_sums
+  # create column for irrigation status
+  dem_sums$irr_count <- NA_integer_
+  # determine whether irrigated area is larger than rainfed area, assign status (irrigated or not)
+  dem_sums[1,3] <- if_else(dem_sums[1,2] > dem_sums[13,2], TRUE, NA)
```

Comment:
you knew I was gonna hate this part.... `tidyr` is your friend here... (`separate` and `spread`)

```
dem_sums %>% 
    mutate(name = if_else(grepl("root_tuber", name), gsub("t_t", "tt", name), name)) %>% 
    separate(name, into = c("crop", "water"), sep = "_") %>% 
    spread(water, value) %>% 
    mutate(irr_count = if_else(irr > rfd, TRUE, FALSE))
```

`root_tuber` is a real bitch here cos it has that pesky underscore (otherwise it's a clean separation of crop from irrigation/rfd), so I've basically caught it out and replaced with `roottuber`... make sure to update the next chunk of code so that it looks for `root tuber` too.